usd/include/sdf/usd/usd_parser/USDTransforms.hh

```diff
@@ -0,0 +1,146 @@
+  //
+  namespace usd
+  {
+    /// \brief This class stores the transforms of a schema
+    /// This might contain scale, translate or rotation operations
+    /// The booleans are used to check if there is a transform defined
+    /// in the schema
+    /// Rotation is splitted in a vector because this might be defined
+    /// as a rotation of 3 angles (ZYX, XYZ, etc).
+    class IGNITION_SDFORMAT_USD_VISIBLE UDSTransforms
+    {
+      /// \brief Default constructor
+      public: UDSTransforms();
+
+      /// \brief Translate
+      public: ignition::math::Vector3d Translate();
+
+      /// \brief Scale
+      public: ignition::math::Vector3d Scale();
+
+      /// \brief Rotation
+      public: std::vector<ignition::math::Quaterniond> Rotations();
+
+      /// \brief Set translate
+      public: void SetTranslate(const ignition::math::Vector3d &_translate);
+
+      /// \brief Set scale
+      public: void SetScale(const ignition::math::Vector3d &_scale);
+
+      /// \brief Add rotation
+      public: void AddRotation(const ignition::math::Quaterniond &_q);
+
+      /// \brief True if there is a rotation ZYX defined or false otherwise
+      bool RotationZYX();
+
+      /// \brief True if there is a rotation XYZ defined or false otherwise
+      bool RotationXYZ();
+
+      /// \brief True if there is a rotation (as a quaterion) defined
+      /// or false otherwise
+      bool Rotation();
+
+      /// \brief Set if there is any rotation ZYX defined
+      void SetRotationZYX(bool _rotationZYX);
+
+      /// \brief Set if there is any rotation XYZ defined
+      void SetRotationXYZ(bool _rotationXYZ);
+
+      /// \brief Set if there is any rotation defined
+      void SetRotation(bool _rotation);
+
+      /// \brief Private data pointer.
+      IGN_UTILS_IMPL_PTR(dataPtr)
+    };
+
+    /// \brief This function will parse all the parents transforms of a prim
+    /// This will stop when the name of the parent is the same as _schemaToStop
+    /// \param[in] _prim Initial prim to read the transform
+    /// \param[in] _usdData USDData structure to get info about the prim, for
+    /// example: metersperunit
+    /// \param[out] _tfs A vector with all the transforms
+    /// \param[out] _scale The scale of the prims
+    /// \param[in] _schemaToStop Name of the prim where the loop will stop
+    /// reading transforms
+    void GetAllTransforms(
+      const pxr::UsdPrim &_prim,
+      USDData &_usdData,
+      std::vector<ignition::math::Pose3d> &_tfs,
+      ignition::math::Vector3d &_scale,
+      const std::string &_schemaToStop);
+
+    /// \brief This function get the transform from a prim to the specified
+    /// schemaToStop variable
+    /// This will stop when the name of the parent is the same as _schemaToStop
+    /// \param[in] _prim Initial prim to read the transform
+    /// \param[in] _usdData USDData structure to get info about the prim, for
+    /// example: metersperunit
+    /// \param[out] _pose Pose of the prim
```

Comment:
Can you document what this pose is relative to? Is the pose expressed from `_schemaToStop` to `_prim`, or from `_prim` to `_schemaToStop`?